src/components/TVChartContainer/TVChartContainer.tsx

```diff
@@ -0,0 +1,142 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { TV_SAVE_LOAD_CHARTS } from "config/localStorage";
+import { useLocalStorage, useMedia } from "react-use";
+import { defaultChartProps, disabledFeaturesOnMonile, SaveLoadAdapter } from "./constants";
+import useTVDatafeed from "domain/tradingview/useTVDatafeed";
+import { IChartingLibraryWidget, IPositionLineAdapter } from "./charting_library/charting_library";
+import { getPeriodFromResolutions, supportedResolutions } from "domain/tradingview/helper";
+
+type Props = {
+  symbol: string;
+  chainId: number;
+  savedShouldShowPositionLines: boolean;
+  currentPositions: any[];
+  currentOrders: any[];
+  onSelectToken: () => void;
+  period: string;
+  setPeriod: (period: string) => void;
+};
+
+export default function TVChartContainer({
+  symbol,
+  chainId,
+  savedShouldShowPositionLines,
+  currentPositions,
+  currentOrders,
+  onSelectToken,
+  period,
+  setPeriod,
+}: Props) {
+  const chartContainerRef = useRef<HTMLDivElement | null>(null);
+  const tvWidgetRef = useRef<IChartingLibraryWidget | null>(null);
+  const [chartReady, setChartReady] = useState(false);
+  const [tvCharts, setTvCharts] = useLocalStorage(TV_SAVE_LOAD_CHARTS, []);
+  const datafeed = useTVDatafeed();
+  const isMobile = useMedia("(max-width: 550px)");
+
+  const drawLineOnChart = useCallback(
+    (title: string, price: number) => {
+      if (chartReady && tvWidgetRef.current?.activeChart?.().dataReady()) {
+        return tvWidgetRef.current
+          .activeChart()
+          .createPositionLine({ disableUndo: true })
+          .setText(title)
+          .setPrice(price)
+          .setQuantity("")
+          .setLineStyle(1)
+          .setLineLength(1)
+          .setBodyFont(`normal 12pt "Relative", sans-serif`)
+          .setBodyTextColor("#fff")
+          .setLineColor("#3a3e5e")
+          .setBodyBackgroundColor("#3a3e5e")
+          .setBodyBorderColor("#3a3e5e");
+      }
+    },
+    [chartReady]
+  );
+
+  useEffect(() => {
```

Comment:
I think it would be more readable to give a name for function inside effect, e.g
```
useEffect(function updateLines() {...}, [...])
```

Reply:
Done!